modules/RTC/JitsiLocalTrack.js

```diff
@@ -163,6 +163,9 @@ export default class JitsiLocalTrack extends JitsiTrack {
         // correspond to the id of a matching device from the available device list.
         this._realDeviceId = this.deviceId === '' ? undefined : this.deviceId;
 
+        // The source name that will be signaled for this track.
+        this._sourceName = null;
```

Comment:
I think it may have to be the conference or a "sender" nested inside the conference that remembers which source name is used for which local tracks, because `JitsiLocalTrack` is technically allowed to be in multiple "conferences" or at least peer connection (it may be the case for Spot, but will have to check that). Anyway lets try to start this way.

Reply:
I don't understand why sources with the same source names cannot be shared across multiple peerconnections ?

Reply:
That's not what I meant. The thing is the `JitsiLocalTrack` when created does not belong to any conference and I thought that one can participate in multiple `JitsiConference` instances, but after checking that is not the case, because there's `conference` field which allows only one. I thought that maybe Spot was doing that for proxying between the peerconnections, but I think it uses raw MediaStreamTrack to do that.